File: android/src/main/java/com/gometroapp/uma/sample/MainActivity.java
package com.gometroapp.uma.sample;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.gometroapp.mobile.android.logging.AndroidLogger;
import com.gometroapp.mobile.core.logging.LoggerFactory;
import com.gometroapp.uma.GoMetroUma;

import android.provider.Settings;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected void onResume() {
        super.onResume();

        String deviceID = Settings.Secure.getString(getContentResolver(),
            Settings.Secure.ANDROID_ID);

        GoMetroUma.initialise(
            this,
            BuildConfig.GOMETRO_UMA_USERNAME,
            BuildConfig.GOMETRO_UMA_PASSWORD,
            deviceID
        );

        LoggerFactory.setLogger(new AndroidLogger(true, "GoMetroUma"));
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        GoMetroUma.onActivityResult(requestCode, resultCode, data);
    }
}

File: android/src/main/java/com/gometroapp/uma/sample/MainActivity.kt
package com.gometroapp.uma.sample

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.gometroapp.uma.client.GoMetroUma
import com.gometroapp.uma.internal.TAG
import io.reactivex.disposables.CompositeDisposable

class MainActivity : AppCompatActivity() {

  private lateinit var goMetroUma: GoMetroUma

  private val disposable = CompositeDisposable()

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    // get GoMetroUma instance
    goMetroUma = GoMetroUma.shared
  }

  override fun onResume() {
    super.onResume()

    // initialize the GoMetro SDK
    disposable.add(
      goMetroUma.init(
        this,
        MainActivity::class.java.canonicalName ?: "",
        "this-is-a-token"
      )
        .subscribe({
          Log.i(TAG, "SM and Mobidot SDKs started")
        }, {
          Log.e(TAG, it.message!!)
        })
    )
  }

  override fun onRequestPermissionsResult(
    requestCode: Int,
    permissions: Array<out String>,
    grantResults: IntArray
  ) {
    super.onRequestPermissionsResult(requestCode, permissions, grantResults)
    goMetroUma.onRequestPermissionsResult(
      requestCode,
      permissions,
      grantResults
    )
  }

  override fun onDestroy() {
    super.onDestroy()
    disposable.dispose()
  }
}
